dts/bindings/i2c/zephyr,i2c-emul-controller.yaml

```diff
@@ -10,6 +10,11 @@ include: i2c-controller.yaml
 properties:
   reg:
     required: true
+  target-pio-mode:
+    type: boolean
+    description: |
+      This option is used when the I2C target is enabled and it can support PIO
+      (Programmed I/O) mode for I2C target transfer.
```

Comment:
```suggestion
      This option is used when the I2C target is enabled and it can support buffered
      mode for I2C target transfer.
```

Reply:
Leaving this unresolved just so I don't lose track of this change if we want to default to pio mode.